modules/completion/init.zsh

```diff
@@ -15,7 +15,15 @@ fi
 fpath=("${0:h}/external/src" $fpath)
 
 # Load and initialize the completion system ignoring insecure directories.
-autoload -Uz compinit && compinit -i
+# Will only rebuild ~/.zcompdump once per day
+autoload -Uz compinit
+if [ $(date +'%j') != $(stat -f '%Sm' -t '%j' ~/.zcompdump) ]; then
```

Comment:
So, here are my thoughts:

First, `date +'%j'` should be replaced with `date -I`.  Not only could relying on the `%j` formatting string from [`strftime`](https://developer.apple.com/legacy/library/documentation/Darwin/Reference/ManPages/man3/strftime.3.html#//apple_ref/doc/man/3/strftime) produce a false negative if you came back to your terminal exactly a year after the last time you used it, but this also makes it easier to compare the output to the GNU coreutils version of stat.  See below.

I think for Mac, the second half of that comparison command should be modified to `stat -f '%Sm' -t '%F' ~/.zcompdump` in accordance with the above change.

For Linux, at least the latest Ubuntu image in Docker, the version of stat being used by default is `stat (GNU coreutils) 8.25`.  For this version of stat, the appropriate command to get the data is `stat -c %y ~/.zcompdump | cut -d ' ' -f1`

The biggest question I have is how we should tell between them.  You might first be tempted to just check whether you're on Mac or Linux with `uname -s` or something, but I have the GNU coreutils installed on my Mac via Homebrew that supercedes the BSD version of [`stat`](https://developer.apple.com/legacy/library/documentation/Darwin/Reference/ManPages/man1/stat.1.html) that comes installed by default.  If you were to run the "Mac" version of the `stat` command on my machine, it would fail due to invalid arguments.  So that's out.

The GNU coreutils version of [`stat`](http://www.gnu.org/software/coreutils/manual/html_node/stat-invocation.html#stat-invocation) comes with a `--version` flag you can pass to get the version and copyright information.  The BSD version doesn't have any kind of version flag that I can find.  So we could do a `stat --version` and if the command completes successfully, just assume it's the GNU version, and issue that version of the command.  Depending on what other versions of `stat` exist out there somewhere, we might also want / need to `grep` for the "GNU coreutils" string inside the version info to verify the version match.  But if `stat --version` returns with a non-zero error code, then we could assume it's BSD `stat`, and issue that version of the command.  Regardless of which version ends up running, the results of the command would need to be collected into a variable prior to running the comparison against the output of `date -I`.

Last of all, my `.zcompdump` file is located at `~/.config/zsh/.zcompdump` instead of `~/.zcompdump`.  So instead of hard coding that file location, it would be good to define the folder that file is in as the parent of the Prezto installation folder, wherever that happens to be.  (I know, I know, I have a really weird setup)